storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -607,6 +608,27 @@ protected String javaCmd(String cmd) {
         return ret;
     }
 
+    /**
+     * Extracting out to mock it for tests.
+     * @return true if on Linux.
+     */
+    protected boolean isOnLinux() {
+        return SystemUtils.IS_OS_LINUX;
+    }
+
+    private void addNumaPinningIfApplicable(String numaId, List<String> commandList) {
```

Comment:
It can be confusing since we are actually adding `numactl` as a prefix. This function looks like we are adding `numactl` to the end of the `commandList`. Although when we use it, `commandList` is empty.